src/core/EdFi.DataManagementService.Core/Handler/GetByResourceNameHandler.cs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.DataManagementService.Core.Backend;
+using EdFi.DataManagementService.Core.External.Backend;
+using EdFi.DataManagementService.Core.External.Interface;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using Microsoft.Extensions.Logging;
+using static EdFi.DataManagementService.Core.External.Backend.GetResult;
+
+namespace EdFi.DataManagementService.Core.Handler;
+internal class GetByResourceNameHandler(IDocumentStoreRepository _documentStoreRepository, ILogger _logger)
+    : IPipelineStep
+{
+    public async Task Execute(PipelineContext context, Func<Task> next)
+    {
+        _logger.LogDebug("Entering GetByResourceNameHandler - {TraceId}", context.FrontendRequest.TraceId);
+
+        int offset = 0;
+        int limit = 25;
+
+        if (context.FrontendRequest.QueryParameters.ContainsKey("offset"))
```

Comment:
This is all behavior that belongs in a middleware. Trust me, it's going to get bigger and more complicated fast.
I suggest a middleware called `ValidateQueryMiddleware`. For now all it needs to do is validate the special query parameters `limit` and `offset`, and put them into a `PaginationParameters` object on `PipelineContext` if they are valid. We already have the `PaginationParameters` object. 

Example from Meadowlark.js at:
https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/21eca1ca8eeb0362354f94304821eb73cf9910e7/Meadowlark-js/packages/meadowlark-core/src/middleware/ValidateQueryMiddleware.ts#L30

which calls: 

https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/21eca1ca8eeb0362354f94304821eb73cf9910e7/Meadowlark-js/packages/meadowlark-core/src/validation/PaginationValidator.ts#L16